[ET-VK][Ops] aten.clamp, aten.hardtanh, aten.relu

junpi3 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #2488
* __->__ #2446
* #2445

EZ operator for warmup. The most complicated logic is merging them.
```
- func: clamp(Tensor self, Scalar? min=None, Scalar? max=None) -> Tensor
- func: hardtanh(Tensor self, Scalar min_val=-1, Scalar max_val=1) -> Tensor
- func: relu(Tensor self) -> Tensor
```

This change is a sample of what adding operators to ET-VK will look like.

It copies the core logic from [`clamp.glsl`](https://github.com/pytorch/pytorch/blob/cceabe873f11c6611f627a3bb0055994952ec6b8/aten/src/ATen/native/vulkan/glsl/clamp.glsl) and [`Clamp.cpp`](https://github.com/pytorch/pytorch/blob/cceabe873f11c6611f627a3bb0055994952ec6b8/aten/src/ATen/native/vulkan/ops/Clamp.cpp) into the new framework.

Differential Revision: [D54886406](https://our.internmc.facebook.com/intern/diff/D54886406/)

Commit:
[ET-VK][Ops] aten.clamp, aten.hardtanh, aten.relu

EZ operator for warmup. The most complicated logic is merging them.
```
- func: clamp(Tensor self, Scalar? min=None, Scalar? max=None) -> Tensor
- func: hardtanh(Tensor self, Scalar min_val=-1, Scalar max_val=1) -> Tensor
- func: relu(Tensor self) -> Tensor
```

This change is a sample of what adding operators to ET-VK will look like.

It copies the core logic from [`clamp.glsl`](https://github.com/pytorch/pytorch/blob/cceabe873f11c6611f627a3bb0055994952ec6b8/aten/src/ATen/native/vulkan/glsl/clamp.glsl) and [`Clamp.cpp`](https://github.com/pytorch/pytorch/blob/cceabe873f11c6611f627a3bb0055994952ec6b8/aten/src/ATen/native/vulkan/ops/Clamp.cpp) into the new framework.

Differential Revision: [D54886406](https://our.internmc.facebook.com/intern/diff/D54886406/)

[ghstack-poisoned]

Commit:
Update on "[ET-VK][Ops] aten.clamp, aten.hardtanh, aten.relu"


EZ operator for warmup. The most complicated logic is merging them.
```
- func: clamp(Tensor self, Scalar? min=None, Scalar? max=None) -> Tensor
- func: hardtanh(Tensor self, Scalar min_val=-1, Scalar max_val=1) -> Tensor
- func: relu(Tensor self) -> Tensor
```

This change is a sample of what adding operators to ET-VK will look like.

It copies the core logic from [`clamp.glsl`](https://github.com/pytorch/pytorch/blob/cceabe873f11c6611f627a3bb0055994952ec6b8/aten/src/ATen/native/vulkan/glsl/clamp.glsl) and [`Clamp.cpp`](https://github.com/pytorch/pytorch/blob/cceabe873f11c6611f627a3bb0055994952ec6b8/aten/src/ATen/native/vulkan/ops/Clamp.cpp) into the new framework.

Differential Revision: [D54886406](https://our.internmc.facebook.com/intern/diff/D54886406/)

[ghstack-poisoned]

SS-JIA reviewed (approved):
Generally LGTM. Left some comments on Phabricator

Commit:
Update on "[ET-VK][Ops] aten.clamp, aten.hardtanh, aten.relu"


EZ operator for warmup. The most complicated logic is merging them.
```
- func: clamp(Tensor self, Scalar? min=None, Scalar? max=None) -> Tensor
- func: hardtanh(Tensor self, Scalar min_val=-1, Scalar max_val=1) -> Tensor
- func: relu(Tensor self) -> Tensor
```

This change is a sample of what adding operators to ET-VK will look like.

It copies the core logic from [`clamp.glsl`](https://github.com/pytorch/pytorch/blob/cceabe873f11c6611f627a3bb0055994952ec6b8/aten/src/ATen/native/vulkan/glsl/clamp.glsl) and [`Clamp.cpp`](https://github.com/pytorch/pytorch/blob/cceabe873f11c6611f627a3bb0055994952ec6b8/aten/src/ATen/native/vulkan/ops/Clamp.cpp) into the new framework.

Differential Revision: [D54886406](https://our.internmc.facebook.com/intern/diff/D54886406/)

[ghstack-poisoned]

Commit:
Update on "[ET-VK][Ops] aten.clamp, aten.hardtanh, aten.relu"


EZ operator for warmup. The most complicated logic is merging them.
```
- func: clamp(Tensor self, Scalar? min=None, Scalar? max=None) -> Tensor
- func: hardtanh(Tensor self, Scalar min_val=-1, Scalar max_val=1) -> Tensor
- func: relu(Tensor self) -> Tensor
```

This change is a sample of what adding operators to ET-VK will look like.

It copies the core logic from [`clamp.glsl`](https://github.com/pytorch/pytorch/blob/cceabe873f11c6611f627a3bb0055994952ec6b8/aten/src/ATen/native/vulkan/glsl/clamp.glsl) and [`Clamp.cpp`](https://github.com/pytorch/pytorch/blob/cceabe873f11c6611f627a3bb0055994952ec6b8/aten/src/ATen/native/vulkan/ops/Clamp.cpp) into the new framework.

Differential Revision: [D54886406](https://our.internmc.facebook.com/intern/diff/D54886406/)

[ghstack-poisoned]
